PowerEditor/src/ScitillaComponent/AutoCompletion.cpp

```diff
@@ -46,15 +46,65 @@ static bool isAllDigits(const generic_string &str)
 	return std::all_of(str.begin(), str.end(), ::isdigit);
 }
 
+bool AutoCompletion::addLanguageCustomChars()
+{
+	if (_funcCalltip._additionalWordChar.empty())
+		return false;
+
+	string charList;
+	auto defaultCharListLen = _pEditView->execute(SCI_GETWORDCHARS);
+	char *defaultCharList = new char[defaultCharListLen + 1];
+	_pEditView->execute(SCI_GETWORDCHARS, 0, reinterpret_cast<LPARAM>(defaultCharList));
+	defaultCharList[defaultCharListLen] = '\0';
+	charList = defaultCharList;
+	delete[] defaultCharList;
+
+	string additionalWordChar = std::string(_funcCalltip._additionalWordChar.begin(), _funcCalltip._additionalWordChar.end());
+	string chars2addStr;
+	for (size_t i = 0; i < additionalWordChar.length(); ++i)
+	{
+		bool found = false;
+		char char2check = additionalWordChar[i];
+		for (size_t j = 0; j < charList.length(); ++j)

```

Comment:
I'll take a look at this but originally this looping structure was copied from ScintillaEditView.cpp for consistency.